### DISCUSSION TITLE
Floating Windows for one Application and all Childs

### DISCUSSION THREAD

**Discussion by k-laig** [i3/i3#6135] [Q&A] [answered]
Hey, I am currently using Waveform 13 on debian 12, with i3 and no session manager, and Waveform it is designed for a stacking wm, there are many popups that are not displayed correctly. Even worse, many plugins just glitch out in the scaling and crash completly when switching from tiling to floating. So I wondered if there is a way to just make all new windows that are drawn for this app floating windows from the start. The plugins are sandboxed, so they have different Apllication ID's. Thx for your help, and please be patient with me, as I setup i3 a few years ago, and haven't touch my dotfiles for so long.

**Comment by orestisfl** [comment, marked as answer]
See https://i3wm.org/docs/userguide.html#for_window

```
for_window [class=Waveform] floating enable
```

But you need to find the exact class Waveform is using with `xprop WM_CLASS` and click on a window (it's the second item in the list)

**Reply by k-laig** [reply]
Thank you, I will try this.

**Reply by k-laig** [reply]
This is working. Thank you for your time, I really could have read the docs more careful.
